web/src/lib/utils/slideshow-history.ts

```diff
@@ -0,0 +1,40 @@
+export class SlideshowHistory {
+  private slideshowHistory: string[] = [];
+  private slideshowHistoryIndex = 0;
+
+  constructor(private onNavigate: (assetId: string) => void) {}
+
+  reset(assetId: string) {
+    this.slideshowHistory = [assetId];
+    this.slideshowHistoryIndex = 0;
+  }
+
+  append(assetId: string) {
```

Comment:
I think `queue` might make more sense. I know in a playlist you can queue items to play after the next one finishes. This seems like a similar concept, basically a playlist.

Reply:
I like this analogy with the playlist

Reply:
I like the slideshow changes, maybe it would make sense to re-name this file/class to `Slideshow` which itself has a history, so you have `slideshow.history` instead of `slideshowHistory.history`.